chapter_5_practic/task12.cpp

```diff
@@ -0,0 +1,100 @@
+// Revise the four - function fraction calculator from Exercise 12, Chapter 4, so that it uses
+// functions for each of the four arithmetic operations.They can be called fadd(), fsub(),
+// fmul(), and fdiv().Each of these functions should take two arguments of type struct
+// fraction, and return an argument of the same type.
+
+#include <iostream>
+
+struct fraction
+{
+	int nominator = 0;
+	int denominator = 1;
+};
+
+fraction fadd(fraction, fraction);
+fraction fsub(fraction, fraction);
+fraction fmul(fraction, fraction);
+fraction fdiv(fraction, fraction);
+void output(fraction a);
+
+fraction first;
+fraction second;
+fraction third;
+
+int main()
+{
+	char repeat = 'y';
+
+	while (repeat == 'y')
+	{
+		char i = ' ';
+		char sign = ' ';
+
+		std::cout << "Enter first fraction : for example 1/2" << std::endl;
+		std::cin >> first.nominator >> i >> first.denominator;
+		std::cout << "Enter operation '+', '-', '*', '/' ";
+		std::cin >> sign;
+		std::cout << "Enter second fraction : for example 2/5" << std::endl;
+		std::cin >> second.nominator >> i >> second.denominator;
+
+		switch (sign) // �����������
+		{
+		case '+':
+			output(fadd(first, second));
+			break;
+		case '-':
+			output(fsub(first, second));
+			break;
+		case '*':
+			output(fmul(first, second));
+			break;
+		case '/':
+			output(fdiv(first, second));
+			break;
+		default:
+			std::cout << "Warning: \"" << sign << "\" is unknown operator" << std::endl;
+		}
+
+		std::cout << "Do another(y / n) ? "; // ������� ��������� ��� ���
+		std::cin >> repeat;
+	}
+
+	return EXIT_SUCCESS;
+}
+
+fraction fadd(fraction first, fraction second)
+{
+	fraction temp;
```

Comment:
prefer const-correctness, if possible
I propose to write in the following way:
`int const numerator = first.nominator * second.denominator + first.denominator * second.nominator;
int const denominator = first.denominator * second.denominator;
return fraction{numerator,denominator };`

Reply:
here and below

Reply:
https://github.com/Dmitriy-Romanchuk/master/pull/5/commits/381b0809f897ecb9e1ba28e1619999d4b570e78f
But I'm not sure if I did it right